constraint.go

```diff
@@ -138,6 +202,18 @@ func prereleaseCheck(v, c *Version) bool {
 // Constraint functions
 //-------------------------------------------------------------------
 
+type operator rune
+
+const (
+	equal            operator = '='
+	notEqual         operator = '≠'
+	greaterThan      operator = '>'
+	lessThan         operator = '<'
+	greaterThanEqual operator = '≥'
+	lessThanEqual    operator = '≤'
+	pessimistic      operator = '~'
```

Comment:
Not sure if this is a good idea, but have you considered making of these strings, in case in a far future we want to be writing//rewriting these operators out to a file ?

Reply:
I have considered that, but I concluded that exposing the operators would make them a public API and I wasn't sure what that API should really look like - e.g. whether `equal` should be empty string or `=`. i.e. there were some edge cases that I didn't want to decide, and so `rune` makes it more obvious that this decision is yet to be made and it's more consciously _not_ meant to be used directly as a public API, but purely for comparison.

You could argue we are already making it public API in the context of `sort.Interface`, as turning runes to strings would likely affect order. However, I still think that displaying operators somewhere is a separate use case and so the potential future duplication (each operator having its own rune + string representation) is fine.

Reply:
Sounds good to me 🙂